app/src/main/java/com/firebase/uidemo/database/ChatActivity.java

```diff
@@ -108,10 +106,10 @@ public void onStart() {
         // Default Database rules do not allow unauthenticated reads, so we need to
         // sign in before attaching the RecyclerView adapter otherwise the Adapter will
         // not be able to read any data from the Database.
-        if (!isSignedIn()) {
-            signInAnonymously();
-        } else {
+        if (isSignedIn()) {
             attachRecyclerViewAdapter();
+        } else {
```

Comment:
Same as before: this change seems unrelated to the topic of this PR and just expresses a personal preference. While that preference might well be better than the existing code, it should not pollute this PR.